src/renderer/rendered_window.rs

```diff
@@ -232,48 +164,112 @@ impl RenderedWindow {
         }
 
         {
-            if 1.0 - self.scroll_t < std::f32::EPSILON {
-                // We are at destination, move t out of 0-1 range to stop the animation.
-                self.scroll_t = 2.0;
-                self.snapshots.clear();
+            let scroll_destination = 0.0;
+            let zeta = 1.0;
+            let omega = 4.0 / (zeta * settings.scroll_animation_length);
+            let k_p = omega * omega;
+            let k_d = -2.0 * zeta * omega;
+            let acc = k_p * (scroll_destination - self.current_scroll) + k_d * self.scroll_v;
+            self.scroll_v += acc * dt;
+            self.current_scroll += self.scroll_v * dt;
+
+            if (self.current_scroll - scroll_destination).abs() < 0.01 {
+                self.reset_scroll();
             } else {
                 animating = true;
-                self.scroll_t = (self.scroll_t + dt / settings.scroll_animation_length).min(1.0);
             }
-
-            self.current_scroll = ease(
-                ease_out_expo,
-                self.start_scroll,
-                self.scroll_destination,
-                self.scroll_t,
-            );
         }
 
         animating
     }
 
+    pub fn draw_surface(
+        &mut self,
+        canvas: &mut Canvas,
+        pixel_region: &Rect,
+        font_dimensions: Dimensions,
+        default_background: Color,
+    ) {
+        let scroll_offset_lines = self.current_scroll.floor();
+        let scroll_offset = scroll_offset_lines - self.current_scroll;
+        let scroll_offset_pixels = (scroll_offset * font_dimensions.height as f32).round() as isize;
+        let mut has_transparency = false;
+
+        let lines: Vec<(Matrix, &Line)> = (0..self.grid_size.height as isize + 1)
+            .filter_map(|i| {
+                let line_index = (self.scrollback_top_index + scroll_offset_lines as isize + i)
+                    .rem_euclid(self.scrollback_lines.len() as isize)
+                    as usize;
+                if let Some(line) = &self.scrollback_lines[line_index] {
+                    let mut m = Matrix::new_identity();
+                    m.set_translate((
+                        pixel_region.left(),
+                        pixel_region.top()
+                            + (scroll_offset_pixels + (i * font_dimensions.height as isize)) as f32,
+                    ));
+                    Some((m, line))
+                } else {
+                    None
+                }
+            })
+            .collect();
+
+        let mut background_paint = Paint::default();
+        background_paint.set_blend_mode(BlendMode::Src);
+        background_paint.set_alpha(default_background.a());
+
+        let save_layer_rec = SaveLayerRec::default()
+            .bounds(pixel_region)
+            .paint(&background_paint);
+        canvas.save_layer(&save_layer_rec);
+        canvas.clear(default_background.with_a(255));
```

Comment:
We might need some macOS specific code here, because your code does not work at all on Wayland for example, but I guess it's the same on all other platforms, only the current line turns transparent when you set `g:neovide_transparency=false`

Unfortunately I don't have a macOS to test myself, it would be best if someone that can test both platforms would do the required modifications.

I also don't understand how macOS works, and why we need this https://github.com/neovide/neovide/blob/main/src/window/draw_background.rs.

On the other platforms it's enough to just render the background as transparent.

Note that my code sets the alpha to fully opaque at this point, because we are doing the blending in the end in the `draw` function, so that it applies to all windows, including floating windows, that could already have alpha applied.

Reply:
Another thing to keep in mind is that the transparency code is extremely sensitive to changes. Skia is a big black box that dislikes some things very much, for example if you call `canvas.draw_picture` with a blend mode it's really slow. See https://github.com/neovide/neovide/pull/1790#discussion_r1141520457

In the first versions of my code there was also problems with the background blur on some systems, some simplification of the code fixed that, but it's unclear what it actually was.

Reply:
I usually use macOS, but I also have a Win11 environment, so I will test it.
...... once my pc comes back from repairs!
(I think it should be home tomorrow or the day after......)

Reply:
When I tested it, it did not cause any problematic behavior at a glance on either macos or windows. If you don't mind, could you further elaborate on what problem you are anticipating......?

Reply:
@AsPulse,

If I run `neovide --multigrid` in KDE Plasma Wayland, with your changes, and set `let g:neovide_transparency=0.5` I see this.

![changes](https://github.com/neovide/neovide/assets/9946255/17d90268-3544-4af7-9b4d-fe9418107d12)

Notice only the edited lines are transparent and the borders around the grid.

And without the changes it looks like this
![original](https://github.com/neovide/neovide/assets/9946255/8eb15159-6817-4967-97c1-16a47d82aaaa)

It's not as good as it could be due to the triangles of the statusline being drawn with the foreground color. But that's an issue with the theme/lualine configuration.

Reply:
Hmmm ......
Even with the modified code, if you force to trigger a redraw, for example by resizing the window, doesn't it draw correctly?

I am encountering the problem that even in the before modifying code (even in the `main` branch if you want), "only the modified line reflects the transparency".
I am not sure why, and it may or may not reproduce after several restarts. ......
So I have been using it for a long time with a slightly different window size, like .......

Therefore I'm not sure it stems from a code change ...... (although it is possible that the code change may have made it more prone to this problem).
I would be glad if you could try the window resizing to help me isolate that.